lib/logger.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file Abstracted logging properties and methods.
+ *
+ * If for any reason, they need a different logging tool or need to be extended,
+ * the changes can be made here.
+ */
+var merge = require('lodash.merge');
+var util = require('util');
+var winston = require('winston');
+
+/**
+ * Abstracts fully-fleshed logging module to the logger property of this module.
+ *
+ * If logging with metadata, or requiring winston.log's more advanced features,
+ * use exports.logger.log() instead of exports.log().
+ * @example
+ *   exports.logger(options).log('info', 'Test Log Message', { anything: 'This is metadata' });
+ *
+ * @param {Object} options - Configuration options.
+ * @return {Object} An instance of Winston.
+ */
+exports.logger = function (options) {
+  'use strict';
+
+  options = getOptions(options);
+
+  var exitOnError = options.logger.exitOnError;
+  var level = options.logger.level;
+  var logFile = options.logger.logFile;
+  var silent = options.logger.silent;
+  var transports = [
+    new winston.transports.Console({
```

Comment:
cc: @rebmullin @sergesemashko @conortm @gvorbeck @beynya @nikkiana

I'm all for that @e2tha-e (new default configs). We have a [task in rally](https://rally1.rallydev.com/#/26707410139ud/detail/task/41508812070) which was meant to help us look into *existing* configuration systems for node modules. @sergesemashko are you working on that task?

The "standards" that I've been pushing for here is merely to uphold a consistency across all files. I realize that these are not done in the best way - but until we have an official way I want all of the files consistent.

I have always been in favor of finding an existing system for setting up the configurations on these scripts from the many, many node-project configuration systems which already exist.

As for default options being set only once - please note that each file needs to house their own default options. Just about each script is made to be used completely on their own outside of pattern-library-utilities. So whatever we choose, I request that "default options" for an individual script are housed inside that individual script. This allows us to:

1. use the file outside of this repo
2. remove or replace or rebuild the file without having to dismantle/change/upgrade too much from the rest of the app

Whatever we choose will need to be compatible with a [typical pattern library's configuration](https://github.com/pattern-library/pattern-library/blob/master/config.yml).